src/VisualStudio/Core/Def/Implementation/Experimentation/AnalyzerVsixSuggestedActionCallback.cs

```diff
@@ -0,0 +1,234 @@
+                var span = currentTime - lastTriggeredTime;
+                if (span.TotalDays >= 1)
+                {
+                    options = options.WithChangedOption(AnalyzerABTestOptions.LastDayUsedSuggestionAction, currentTime.ToBinary());
+                    var usageCount = options.GetOption(AnalyzerABTestOptions.UsedSuggestedActionCount);
+                    usageCount++;
+                    options = options.WithChangedOption(AnalyzerABTestOptions.UsedSuggestedActionCount, usageCount);
+
+                    if (usageCount >= 3)
+                    {
+                        isCandidate = true;
+                        options = options.WithChangedOption(AnalyzerABTestOptions.HasMetCandidacyRequirements, true);
+                    }
+                    _workspace.Options = options;
+                }
+            }
+
+            // If the user still isn't a candidate, then return. Otherwise, we move to checking if the infobar should be shown
+            if (!isCandidate)
+            {
+                lock (_scheduleLock)
+                {
+                    _checksRunning = false;
+                }
+
+                return;
+            }
+
+            ShowInfoBarIfNecessary();
+        }
+
+        private void ShowInfoBarIfNecessary()
+        {
+            if (_experimentationService.IsExperimentEnabled(AnalyzerEnabledFlight))
+            {
+                // If we got true from the experimentation service, then we're in the treatment group, and the experiment is enabled.
+                // We determine if the infobar has been displayed in the past 24 hours. If it hasn't been displayed, then we do so now.
+                var lastDayInfoBarShown = DateTime.FromBinary(_workspace.Options.GetOption(AnalyzerABTestOptions.LastDayInfoBarShown));
+                var utcNow = DateTime.UtcNow;
+                var timeSinceLastShown = utcNow - lastDayInfoBarShown;
+
+                if (timeSinceLastShown.TotalDays >= 1)
+                {
+                    _workspace.Options = _workspace.Options.WithChangedOption(AnalyzerABTestOptions.LastDayInfoBarShown, utcNow.ToBinary());
+
+                    var infoBarService = _workspace.Services.GetRequiredService<IInfoBarService>();
+                    infoBarService.ShowInfoBarInGlobalView(
+                        ServicesVSResources.Analyzer_vsix_try_description,
+                        // Install link
+                        new InfoBarUI(title: ServicesVSResources.Analyzer_vsix_hyperlink,
+                                      kind: InfoBarUI.UIKind.HyperLink,
+                                      action: OpenInstallHyperlink),
+                        // Don't show the InfoBar again link
+                        new InfoBarUI(title: ServicesVSResources.Analyzer_vsix_do_not_show_again,
+                                      kind: InfoBarUI.UIKind.Button,
+                                      action: DoNotShowAgain));
```

Comment:
I believe the install fwlink will have the detail, but I'm not sure. @kuhlenh, do we want a separate link for this?